crypto/bio/bss_file.c

```diff
@@ -346,10 +346,22 @@ static long file_ctrl(BIO *b, int cmd, long num, void *ptr)
     return ret;
 }
 
+static long file_pos(BIO *bp)
+{
+    if ((bp->flags & BIO_FLAGS_UPLINK_INTERNAL) != 0)
+        return UP_ftell(bp->ptr);
+    else
+        return ftell((FILE *)bp->ptr);
+}
+
 static int file_gets(BIO *bp, char *buf, int size)
 {
     int ret = 0;
+    long before, after;
 
+    before = file_pos(bp);
+    if (before < 0)
```

Comment:
This is bad idea! ftell is not guaranteed to work on file BIOs.

Reply:
Oh, good to know!
Likely this was the reason for the strange test failures on Windows.